.github/workflows/refresh-lockfiles.yml

```diff
@@ -59,20 +59,30 @@ jobs:
               }
             })
 
+  get_python_matrix:
+    # Determines which Python versions should be included in the matrix used in
+    # the gen_lockfiles job.
+    if: "github.repository == 'SciTools/iris'"
+    runs-on: ubuntu-latest
+    needs: no_clobber
+    outputs:
+      matrix: ${{ steps.get_py.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: get_py
+        run: echo "::set-output name=matrix::$(ls -1 requirements/ci/py??.yml | xargs -n1 basename | cut -c-4 | jq -cnR '[inputs | select(length>0)]')"
+
   gen_lockfiles:
     # this is a matrix job: it splits to create new lockfiles for each
     # of the CI test python versions.
-    # this list below should be changed when covering more python versions
-    # TODO: generate this matrix automatically from the list of available py**.yml files
-    #       ref: https://tomasvotruba.com/blog/2020/11/16/how-to-make-dynamic-matrix-in-github-actions/
     if: "github.repository == 'SciTools/iris'"
     runs-on: ubuntu-latest
-    needs: no_clobber
+    needs: get_python_matrix
     
     strategy:
       matrix:
-        python: ['38']
-    
+        python: ${{ fromJSON(needs.get_python_matrix.outputs.matrix) }}
```

Comment:
The `needs....` syntax checks out against [the documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-defining-outputs-for-a-job), but shouldn't `fromJSON()` sit outside the GHA env variable? I.e:

```suggestion
        python: fromJSON(${{ needs.get_python_matrix.outputs.matrix }})
```

Reply:
Unfortunately, when I try that github complains:
![image](https://user-images.githubusercontent.com/8101163/157274387-9edf4847-60b7-4e67-9389-18e399547f12.png)